_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,579 @@
+year:
+
+* Trino now offers improved performance for a variety of operations, including
+ complex join criteria pushdown to connectors, faster aggregations, faster
+ joins, and better performance for large clusters. We have also implemented
+ improvements specifically for aggregations with filters and for the Glue
+ metastore. In addition, we now support dynamic filtering for various connectors
+ and have faster query planning for the Hive, Delta Lake, Iceberg, MySQL,
+ PostgreSQL, and SQL Server connectors.
+* Along with general performance optimizations, there have been a great deal of
+ query planning optimizations that lead to better performance for specific SQL
+ operators. These include faster `INSERT` queries, improved performance for
+ `LIKE` expressions and highly selective `LIMIT` queries, and enhanced
+ performance and reliability for `INSERT` and `MERGE` operations. We also made
+ performance improvements for `JOIN`, `UNION`, and `GROUP BY` queries, as well
+ as faster planning of queries with `IN` predicates. 
+* There are also optimizations for specific SQL types’ performance, such as
+ string, `DECIMAL`, `MAP` and `ROW` types. We have also made aggregations over 
+ `DECIMAL` columns faster and improved the performance of `ROW` type and
+ aggregation.
+* A last set of improvements come from reading open file formats like ORC and
+ Parquet efficiently. We have improved the speed of reading or writing of all 
+ data types from and to Parquet in general. There were also general performance
+ to ORC types, and now have the ability to write Bloom filters in ORC files. We
+ have also improved performance and efficiency for a wide range of ORC and
+ Parquet-related operations.
+
+These improvements in aggregate are at the core of what makes Trino fast. There
+is no silver bullet you can plug in to speed things up. It takes time, effort,
+and smart changes to improve the speed of various systems.
+
+### Runtime improvements
+
+[Trino upgraded Java versions](https://github.com/trinodb/trino/issues/9876)from
+Java 11 to the next long-term support version of Java, which was Java 17. This
+upgrade improves the latency, memory footprint, startup, ramp up, pause times,
+and shut down times for Trino. Along with these performance enhancements one of the other
+critical elements of improving performance for the Java Virtual Machine is to
+improve the garbage collector. Trino uses the G1 garbage collector which can now
+more efficiently reclaim memory and reduce pause times.
+
+Martin did a benchmark on Trino workloads, and reported seeing anywhere from
+10-15% improvement in speed in Java 17. Many users have also reported a similar
+boost in their workloads, and we have yet to have a user mention any regressions
+related to Java 17, which is exciting news. Aside from having to perform the
+upgrades, we get a lot of these performance enhancements for free.
+
+On top of performance, upgrading to Java 17 adds new Java language features to
```

Comment:
Cut this down to just one sentence as well about language and API improvements now being available for use.. and forget about the vector stuff and more details